### PR TITLE
FEATURE: Ask user to confirm topic deletion when views exceed a certain amount

### PR DESCRIPTION
**Pull request by lis2**
Before deleting a topic that has a high number of views (default of 5000), the user will be prompted with a confirmation popup. This works for all delete buttons on the topic located in: `topic-timeline`, `topic-admin-menu`, `topic-footer-buttons`, and `post-menu` if the post's ID is 1.

The delete button will be disabled while deletion is in progress, to prevent any unwanted behavior.

A site setting is also available to change the minimum amount of views required to display the confirmation popup.

image

discourse-delete-modal-confirm

All kudos are going to @RickyC0626. I only rebased with master and added few qunit tests to ensure that this feature works as expected.

Original PR: https://github.com/discourse/discourse/pull/10459

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/acceptance/topic-test.js [hunk 1]
```diff
@@ -200,15 +200,39 @@ test("Suggested topics", async (assert) => {
   );
 });
 
-skip("Deleting a topic", async (assert) => {
+QUnit.test("Deleting a topic", async (assert) => {
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/tests/acceptance/topic-test.js:203]: This will need to be updated to our new acceptance test format

**Review by eviltrout — CHANGES_REQUESTED** [review]

**Comment by SamSaffron** [comment]
@lis2 happy to get this merged once tests are in the new format!

**Review by eviltrout — APPROVED** [review]

**Comment by SamSaffron** [comment]
Lets get a slightly shorter name for the site setting @lis2 and then go ahead and merge

`min_topic_views_for_delete_confirm` 

Current name is a bit long.

**Comment by RickyC0626** [comment]
Thanks for the help!

**Pull request merged**